src/book.py

```diff
@@ -627,11 +627,160 @@ def _read_bitpanda_pro_trades(self, file_path: Path) -> None:
+                # be implemented if need be.
+                if operation == "transfer":
+                    log.warning(
+                        f"'Transfer' operations are not "
+                        f"implemented, skipping: {file_path} line {row}."
+                    )
+                    continue
+
+                # fail for unknown ops
+                try:
+                    operation = operation_mapping[operation]
+                except KeyError:
+                    raise RuntimeError(f"Unsupported operation '{operation}'")
+
+                if operation == "Deposit":
+                    # This identifies a fiat deposit. Error on yet unknown deposits,
+                    # they will need to be implemented separately
+                    if fiat != "EUR" and asset != "EUR" and asset_class != "fiat":
+                        raise RuntimeError(
+                            "A deposit is expected to be of fiat and 'EUR' currency. "
+                            "Others are not implemented yet."
+                        )
+                    else:
+                        change = misc.force_decimal(amount_fiat)
+                elif operation == "Withdraw":
+                    change = misc.force_decimal(amount_asset)
+                elif operation in ["Buy", "Sell"]:
+                    if asset_price_currency != "EUR":
+                        raise RuntimeError(
+                            "Only Euro is supported as 'Asset market price currency' "
+                            "currency, since price fetching is not fully implemented "
+                            "yet."
+                        )
+                    change = misc.force_decimal(amount_asset)
+                    # Save price in our local database for later.
+                    price = misc.force_decimal(asset_price)
+                    self.price_data.set_price_db(
+                        platform, asset, "EUR", utc_time, price
+                    )
+
+                if change < 0:
+                    raise RuntimeError(
+                        f"Unexpected value for 'Amount Asset' column {change}"
+                    )
+
+                self.append_operation(
+                    operation, utc_time, platform, change, asset, row, file_path
+                )
+
+                if fee != "-":
+                    self.append_operation(
+                        "Fee",
+                        utc_time,
+                        platform,
+                        misc.force_decimal(fee),
+                        fee_currency,
+                        row,
+                        file_path,
+                    )
+
     def detect_exchange(self, file_path: Path) -> Optional[str]:
         if file_path.suffix == ".csv":
             with open(file_path, encoding="utf8") as f:
                 reader = csv.reader(f)
-                header = next(reader, None)
+                csv_content = list(reader)
```

Comment:
Can we accomplish this without reading the whole csv file into memory?

Reply:
sure